subprojects/platform-native/src/main/java/org/gradle/nativeplatform/tasks/InstallExecutable.java

```diff
@@ -184,11 +188,16 @@ protected FileOperations getFileOperations() {
 
     @TaskAction
     public void install() {
-        if (platform.getOperatingSystem().isWindows()) {
-            installWindows();
-        } else {
-            installUnix();
-        }
+        workerLeaseService.withoutProjectLock(new Runnable() {
```

Comment:
Does this need to use the lower-level API because we can't inject `FileOperations` to do the copy?

If so, could you add a comment here saying something to that affect, so the next person knows why we couldn't use the better API.

Reply:
Yep.  I've added a comment.